drivers/misc/qseecom.c

```diff
@@ -3425,6 +3427,19 @@ static int __qseecom_send_cmd(struct qseecom_dev_handle *data,
 		return -ENOENT;
 	}
 
+	if (!memcmp(data->client.app_name, "tzxflattest", strlen("tzxflattest")))
+	{
+		sb = (void *)__qseecom_uvirt_to_kvirt(data,
+						(uintptr_t)req->cmd_req_buf);
+		rb = (void *)__qseecom_uvirt_to_kvirt(data,
+						(uintptr_t)req->resp_buf);
+		if (sb != NULL && req->cmd_req_len >= sizeof(uint32_t) * 2)
+			if (sb[0] != 0x07 || sb[1] != 0x04)
+				sb = NULL;
+		if (sb == NULL)
+			rb = NULL;
```

Comment:
```suggestion
		if (sb != NULL && req->cmd_req_len >= sizeof(uint32_t) * 2) {
			if (sb[0] == 0x07 && sb[1] == 0x04)
				rb = (uint32_t *)__qseecom_uvirt_to_kvirt(data, (uintptr_t)req->resp_buf);
			else
				sb = NULL;
		}
```

Reply:
In my opinion, this suggestion would not make any significant difference with code efficiency as the __qseecom_uvirt_to_kvirt() is simple arithmetic pointer conversion, not involving any memory mapping or whatever, so it's cost even if done unnecessary is nearing zero. I am not sure if this way the code would be more readable - to me it seems not really.
Concerning the cast to uint32_t * instead of void * - I guess that is a matter of taste.
I just followed the pattern already existing in the qseecom source code - you can see there casting to void * even though it is assigned to something else the same way I did it here - in fact, I copied it from the existing code.
So I am not sure what is better - to keep the usage uniform or try to invent something new.
In any case, there is no difference in the end.

Reply:
> __qseecom_uvirt_to_kvirt() is simple arithmetic pointer conversion,

Ok, then yes, doesn't affect efficiency.

> I am not sure if this way the code would be more readable

It also adds the braces for the multi-line if code which is IMO required according to the style guide. But yeah, doesn't matter much

>  Concerning the cast to uint32_t * instead of void * - I guess that is a matter of taste.

Yeah, this is mostly about the adjacent usage of `sizeof(uint32_t)` and the access so one doesn't have to scroll up to the definition to verify this. Again, just minor

Reply:
A nested 'if' is not a problem without braces as long as there is no 'else' there. No 'else' in the original code, no braces. You have 'else' in your suggestion, so you have braces too.